### ISSUE TITLE
A user disabled via OCC causes local 'Internal Server Error' on login.

### ISSUE THREAD

**Issue by FlyinFistOfJuda** [nextcloud/server#2130]
### Steps to reproduce
1. sudo -u apache php /var/www/nextcloud/occ user:disable occtest
2. Disabled User logs in.
3. Internal Server Error is shown on Login Screen for the disabled user and a log is produced in nextcloud.log

### Expected behaviour
A disabled user should not throw an internal server error.

### Actual behaviour
Nextcloud throws an Internal Server Error. See attached nextcloud.log. All other users can work normally and wont see this error.

### Server configuration
**Operating system**:
CentOS 6.8

**Web server:**
MySQL 5.6.

**Database:**
irrelevant

**PHP version:**
PHP 5.6.26 with acpu and redis

**Nextcloud version:** (see Nextcloud admin page)
10.0.1

**Updated from an older Nextcloud/ownCloud or fresh install:**
8.2 OC -> 10.0.1 NC

**Where did you install Nextcloud from:**
Nextcloud Website

**Signing status:**
<details>
<summary>Signing status
```
No errors have been found.
```
</summary>
</details>

**List of activated apps:**
<details>
<summary>App list</summary>
Enabled:
  - activity: 2.3.2
  - admin_audit: 1.0.0
  - comments: 1.0.0
  - dav: 1.0.1
  - federatedfilesharing: 1.0.1
  - federation: 1.0.1
  - files: 1.5.2
  - files_accesscontrol: 1.1.2
  - files_external: 1.0.2
  - files_pdfviewer: 0.8.1
  - files_sharing: 1.0.0
  - files_texteditor: 2.1
  - files_trashbin: 1.0.0
  - files_versions: 1.3.0
  - gallery: 15.0.0
  - notifications: 0.3.0
  - password_policy: 1.0.0
  - provisioning_api: 1.0.0
  - serverinfo: 1.1.1
  - survey_client: 0.1.5
  - systemtags: 1.0.2
  - templateeditor: 0.1
  - theming: 1.0.1
  - updatenotification: 1.0.1
  - workflowengine: 1.0.1
Disabled:
  - encryption
  - external
  - files_automatedtagging
  - files_retention
  - files_videoplayer
  - firstrunwizard
  - user_external
  - user_ldap
  - user_saml

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```
</details>

**The content of config/config.php:**
<details>
<summary>Config report</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or 

Insert your config.php content here
(Without the database password, passwordsalt and secret)
```
'code', 'passwordsalt' => 'code', 'secret' => 'code', 'trusted_domains' => array ( 0 => 'x.x.x.x', 1 => 'hostname', ), 'datadirectory' => '/data1/nextcloud_data', 'filesystem_check_changes' => 1, 'overwrite.cli.url' => 'https://x.x.x.x', 'htaccess.RewriteBase' => '/', 'proxy' => 'proxy:80', 'dbtype' => 'mysql', 'version' => '9.1.1.5', 'dbname' => 'cloud', 'dbhost' => 'localhost', 'dbtableprefix' => 'oc_', 'dbuser' => 'dbuser', 'dbpassword' => 'xxx', 'installed' => true, 'logfile' => '/var/log/nextcloud.log', 'loglevel' => 2, 'logtimezone' => 'Europe/Berlin', 'mail_smtpmode' => 'smtp', 'default_language' => 'de_DE', 'skeletondirectory' => '', 'maintenance' => false, 'mail_from_address' => 'no-reply', 'mail_domain' => 'xxx', 'mail_smtphost' => 'localhost', 'mail_smtpport' => '25', 'asset-pipeline.enabled' => 'false', 'forcessl' => 'true', 'forceSSLforSubdomains' => false, 'appstoreenabled' => false, 'allow_user_to_change_username' => false, 'knowledgebaseenabled' => false, 'updatechecker' => false, 'singleuser' => false, 'theme' => '', 'trashbin_retention_obligation' => 'auto', 'auth.bruteforce.protection.enabled' => false, 'has_internet_connection' => true, 'filelocking.enabled' => 'true', 'memcache.locking' => '\OC\Memcache\Redis', 'memcache.local' => '\OC\Memcache\APCu', 'redis' => array ( 'host' => 'localhost', 'port' => 6379, ), 'debug' => false, );
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...
None

**Are you using encryption:** yes/no
No

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
No

#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>
****
```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your Nextcloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```
</details>

### Client configuration
**Browser:**
Any

**Operating system:**

#### Web server error log
<details>
<summary>Web server error log</summary>
```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log
```
<code>
{"reqId":"8MA6bVAcYuiFcm0AEgWp","remoteAddr":"x.x.x.x","app":"index","message":"Exception: {\"Exception\":\"OC\\\\User\\\\LoginException\",\"Message\":\"Nutzer deaktiviert\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/User\\\/Session.php(291): OC\\\\User\\\\Session->loginWithPassword(*** sensitive parameters replaced ***)\\n#1 \\\/var\\\/www\\\/nextcloud\\\/core\\\/Controller\\\/LoginController.php(237): OC\\\\User\\\\Session->login(*** sensitive parameters replaced ***)\\n#2 [internal function]: OC\\\\Core\\\\Controller\\\\LoginController->tryLogin(*** sensitive parameters replaced ***)\\n#3 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(160): call_user_func_array(Array, Array)\\n#4 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(90): OC\\\\AppFramework\\\\Http\\\\Dispatcher->executeController(Object(OC\\\\Core\\\\Controller\\\\LoginController), 'tryLogin')\\n#5 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/AppFramework\\\/App.php(111): OC\\\\AppFramework\\\\Http\\\\Dispatcher->dispatch(Object(OC\\\\Core\\\\Controller\\\\LoginController), 'tryLogin')\\n#6 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/AppFramework\\\/Routing\\\/RouteActionHandler.php(47): OC\\\\AppFramework\\\\App::main('LoginController', 'tryLogin', Object(OC\\\\AppFramework\\\\DependencyInjection\\\\DIContainer), Array)\\n#7 [internal function]: OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler->__invoke(Array)\\n#8 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/Route\\\/Router.php(293): call_user_func(Object(OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler), Array)\\n#9 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/base.php(983): OC\\\\Route\\\\Router->match('\\\/login')\\n#10 \\\/var\\\/www\\\/nextcloud\\\/index.php(48): OC::handleRequest()\\n#11 {main}\",\"File\":\"\\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/User\\\/Session.php\",\"Line\":465}","level":3,"time":"2016-11-15T10:13:17+01:00","method":"POST","url":"\/login","user":"--"}
</code>
```
</summary>

</details>

#### Browser log
<details>
<summary>Browser log</summary>
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

**Comment by FlyinFistOfJuda** [comment]
Sorry, i totally wasted the ticket structure. Anyhow, the relevant part is the log of nextcloud.log:
`{"reqId":"8MA6bVAcYuiFcm0AEgWp","remoteAddr":"x.x.x.x","app":"index","message":"Exception: {\"Exception\":\"OC\\\\User\\\\LoginException\",\"Message\":\"Nutzer deaktiviert\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/User\\\/Session.php(291): OC\\\\User\\\\Session->loginWithPassword(*** sensitive parameters replaced ***)\\n#1 \\\/var\\\/www\\\/nextcloud\\\/core\\\/Controller\\\/LoginController.php(237): OC\\\\User\\\\Session->login(*** sensitive parameters replaced ***)\\n#2 [internal function]: OC\\\\Core\\\\Controller\\\\LoginController->tryLogin(*** sensitive parameters replaced ***)\\n#3 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(160): call_user_func_array(Array, Array)\\n#4 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(90): OC\\\\AppFramework\\\\Http\\\\Dispatcher->executeController(Object(OC\\\\Core\\\\Controller\\\\LoginController), 'tryLogin')\\n#5 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/AppFramework\\\/App.php(111): OC\\\\AppFramework\\\\Http\\\\Dispatcher->dispatch(Object(OC\\\\Core\\\\Controller\\\\LoginController), 'tryLogin')\\n#6 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/AppFramework\\\/Routing\\\/RouteActionHandler.php(47): OC\\\\AppFramework\\\\App::main('LoginController', 'tryLogin', Object(OC\\\\AppFramework\\\\DependencyInjection\\\\DIContainer), Array)\\n#7 [internal function]: OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler->__invoke(Array)\\n#8 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/Route\\\/Router.php(293): call_user_func(Object(OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler), Array)\\n#9 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/base.php(983): OC\\\\Route\\\\Router->match('\\\/login')\\n#10 \\\/var\\\/www\\\/nextcloud\\\/index.php(48): OC::handleRequest()\\n#11 {main}\",\"File\":\"\\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/User\\\/Session.php\",\"Line\":465}","level":3,"time":"2016-11-15T10:13:17+01:00","method":"POST","url":"\/login","user":"--"}`

**Comment by MorrisJobke** [comment]
will be fixed with #1234

**Comment by MorrisJobke** [comment]
This is not the case anymore with current master.
